src/js/libgeo/GeoOps.js

```diff
@@ -625,6 +625,144 @@ geoOps.PointOnSegment.updatePosition = function(el) {
 };
 geoOps.PointOnSegment.stateSize = 2;
 
+geoOps._helper.PointOnArcCr = function(arc,P){
+    var A = arc.startPoint, B = arc.viaPoint, C = arc.endPoint;
+    //var cr = List.crossratio3harm(P,A,B,C,List.ii);
+    var cr = List.crossratio3harm(A,C,B,P,List.ii);
+    var m = cr.value[0];
+    var n = cr.value[1];
+    if (!CSNumber._helper.isAlmostZero(m)) {
+        n = CSNumber.div(n, m);
+        m = CSNumber.real(1);
+    } else {
+        m = CSNumber.div(m, n);
+        n = CSNumber.real(1);
+    }
+    return [m,n];
+};
+
+geoOps._helper.projectPointToArc = function(arc,P){
+    var conic = arc.matrix;
+    var cen = geoOps._helper.CenterOfConic(conic);
+
+    var diam = List.cross(cen, P);
+ //   Render2D.drawline(diam);
+//    debugger;
+
+    var isec = geoOps._helper.IntersectLC(diam, arc.matrix);
+
+
+    var crA = geoOps._helper.PointOnArcCr(arc, isec[0]);
+    var prodA = CSNumber.mult(crA[0], crA[1]);
+   // console.log("pA",niceprint(prodA));
+    if (crA[0].value.real < 0) prodA = CSNumber.neg(prodA);
+
+    var crB = geoOps._helper.PointOnArcCr(arc,isec[1]);
+    var prodB = CSNumber.mult(crB[0], crB[1]);
+//    console.log("pB",niceprint(prodB));
+    if (crB[0].value.real < 0) prodB = CSNumber.neg(prodB);
+
+    
+    var erg;
+    if(prodA.value.real > 0 && prodB.value.real > 0){
+        console.log("hit1");
+        var d1 = List.projectiveDistMinScal(P, isec[0]);
+        var d2 = List.projectiveDistMinScal(P, isec[1]);
+
+        erg = d1 < d2 ? isec[0] : isec[1];
+    }
```

Comment:
This part is similar to [`PointOnCircle.initialize`](https://github.com/CindyJS/CindyJS/blob/v0.7.3/src/js/libgeo/GeoOps.js#L469-L490). Not sure it would make sense to somehow share this, though.